src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -1056,7 +1056,7 @@ private void CheckRestrictedTypeReceiver(BoundExpression expression, CSharpCompi
                 case BoundKind.Call:
                     {
                         var call = (BoundCall)expression;
-                        if (!call.HasAnyErrors && call.ReceiverOpt != null && (object)call.ReceiverOpt.Type != null)
+                        if (!call.HasAnyErrors && call.ReceiverOpt != null && (object)call.ReceiverOpt.Type != null && !call.Method.GetIsNewExtensionMember())
```

Comment:
>&& !call.Method.GetIsNewExtensionMember() [](http://example.com/codeflow?start=116&length=41)

It is not obvious why new extensions are excluded here.

Reply:
The "receiver" in this BoundCall will not actually be a receiver, it will be passed as an argument, so the receiver restrictions don't apply.
For classic extension method invocation, we also wouldn't do those checks (the receiver in the BoundCall will have been set to null).

```
    [Fact]
    public void GetCompatibleExtension_Conversion_09()
    {
        var src = """
int[] i = default;
i.M();
i.M2();

static class E
{
    extension(System.ReadOnlySpan<int> ros)
    {
        public void M() { }
    }

    public static void M2(this System.ReadOnlySpan<int> ros) { }
}
""";
        var comp = CreateCompilation(src, targetFramework: TargetFramework.Net90);
        comp.VerifyEmitDiagnostics();
    }
```